lms/djangoapps/certificates/api.py

```diff
@@ -228,8 +229,13 @@ def certificate_downloadable_status(student, course_key):
         'is_unverified': True if current_status['status'] == CertificateStatuses.unverified else False,
         'download_url': None,
         'uuid': None,
+        'is_invalid_cert': False,
```

Comment:
@awais786 I see. You explanation make sense. But my request to have the variable `show_generate_cert_btn` to be false when the cert is invalidated still stands.
Upon further thinking, I think I know why I feel the approach on this PR is not ideal.
You put a condition `is_invalid` into function `certificate_downloadable_status` which is meant to return certificate statuses. The condition `is_invalid` has nothing to do with certificate status. It is sourced from another certificate related django model. The student can be invalidated to receive certificate before certificate is ever generated. Therefore returning this fact in this function is not a good design.
I believe it's a good idea to either:

- Expand function `cert_generation_enabled` to include the intended user and take care both the student and instructor case
- Create a new cert api function to return whether cert_generation enabled for the student, and use the new api function to determine whether we should `show_generate_cert_btn` 

The new cert api function maybe better because that can return to a variable you can use in the template.